amazon/ion/ioncmodule.c

```diff
@@ -973,15 +976,18 @@ static iERR ionc_read_timestamp(hREADER hreader, PyObject** timestamp_out) {
     PyDict_SetItemString(timestamp_args, "precision", py_precision);
     BOOL has_local_offset;
     IONCHECK(ion_timestamp_has_local_offset(&timestamp_value, &has_local_offset));
-
     if (has_local_offset) {
         int off_minutes, off_hours;
         IONCHECK(ion_timestamp_get_local_offset(&timestamp_value, &off_minutes));
         off_hours = off_minutes / 60;
         off_minutes = off_minutes % 60;
+        PyObject* py_off_hours = PyInt_FromLong(off_hours);
+        PyObject* py_off_minutes = PyInt_FromLong(off_minutes);
         // Bounds checking is performed in python.
-        PyDict_SetItemString(timestamp_args, "off_hours", PyInt_FromLong(off_hours));
-        PyDict_SetItemString(timestamp_args, "off_minutes", PyInt_FromLong(off_minutes));
+        PyDict_SetItemString(timestamp_args, "off_hours", PyLong_FromLong(off_hours));
+        PyDict_SetItemString(timestamp_args, "off_minutes", PyLong_FromLong(off_minutes));
```

Comment:
Are the final arguments supposed to be `py_off_hours` and `py_off_minutes`?

Did the files you used for memory leak checking include timestamps with offsets?

Reply:
Yeah, Should be `py_off_hours` and `py_off_minutes`.

The test files include them but it doesn't shows memory leak because [PyLong_FromLong() ](https://docs.python.org/3/c-api/long.html#c.PyLong_FromLong) keeps an array for number -5 to 256, so it just returns an exist reference rather than creating a new reference. So there is no memory leak.

By the way, the Travis failed due to python2 issue, this will pass after we merge from main branch.

Reply:
Sounds good. go ahead and merge.